### ISSUE TITLE
Side car not getting injected in the kubernetes pod

### ISSUE THREAD

**Issue by VenkateshSrini** [dapr/dapr#4932]
Hi All,
Sidecar is not getting injected into the pod. The deployment yaml is as below 
`apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    app: dapr-state
  annotations:
    dapr.io/enabled: "true"
    dapr.io/app-id: "dapr-state"
    dapr.io/app-port: "80"
    dapr.io/enable-api-logging: "true"
  name: dapr-state
  namespace: default
spec:
  replicas: 1
  selector:
    matchLabels:
      app: dapr-state
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: dapr-state
    spec:
      containers:
      - image: venkateshsrini3/dapr-state-service
        name: dapr-state-service
        ports:
        - containerPort: 80
        env:
        - name: "State.Store"
          value: "redis-cache-store"`

I see the below log in side car injector
`r-6d5948b65f-tcnv9 -n dapr-system
time="2022-07-25T06:51:56.303449526Z" level=info msg="log level set to: info" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T06:51:56.304431108Z" level=info msg="metrics server started on :9090/" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.metrics type=log ver=1.8.2
time="2022-07-25T06:51:56.305006017Z" level=info msg="starting Dapr Sidecar Injector -- version 1.8.2 -- commit 75982fbcae3541c9a7cbc5095e5438a8da8e89ac" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T06:51:56.310292619Z" level=info msg="Healthz server is listening on :8080" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T06:51:56.325572266Z" level=warning msg="Unable to get SA  UID (%!s(<nil>))" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T06:51:56.325617593Z" level=warning msg="Unable to get SA replicaset-controller:kube-system UID (%!s(<nil>))" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T06:51:56.325626534Z" level=warning msg="Unable to get SA deployment-controller:kube-system UID (%!s(<nil>))" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T06:51:56.325640485Z" level=warning msg="Unable to get SA cronjob-controller:kube-system UID (%!s(<nil>))" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T06:51:56.325647325Z" level=warning msg="Unable to get SA job-controller:kube-system UID (%!s(<nil>))" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T06:51:56.32565277Z" level=warning msg="Unable to get SA statefulset-controller:kube-system UID (%!s(<nil>))" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T06:51:56.325657858Z" level=warning msg="Unable to get SA daemon-set-controller:kube-system UID (%!s(<nil>))" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T06:51:56.325663629Z" level=warning msg="Unable to get SA tekton-pipelines-controller:tekton-pipelines UID (%!s(<nil>))" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
_**time="2022-07-25T06:51:56.325804683Z" level=info msg="Sidecar injector is listening on :4000, patching Dapr-enabled pods" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T07:37:39.561234763Z" level=info msg="AdmissionReview for Kind=/v1, Kind=Pod, Namespace=m2c-steeltoe Name= () UID=96fcfbb2-ed85-457d-90f8-14f8b3eae994 patchOperation=CREATE UserInfo={system:serviceaccount:kube-system:replicaset-controller 5841b366-d28d-40da-af15-4a6c4d7c8e55 [system:serviceaccounts system:serviceaccounts:kube-system system:authenticated] map[]}" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2
time="2022-07-25T07:37:39.562031675Z" level=info msg="Sidecar injector succeeded injection for app ''" instance=dapr-sidecar-injector-6d5948b65f-tcnv9 scope=dapr.injector type=log ver=1.8.2**_`

dapr version :1.8
installed via helm chart
Kubernetes environment - Kubernetes in docker- desktop for winodws
Version of Kubernets: - v1.24.1
version of docker desktop: Docker Desktop 4.10.1 (82475)

sample code is at  [code sample](https://github.com/VenkateshSrini/dapr-state-store-app)
Yaml for deployment at [Yaml for deployment](https://github.com/VenkateshSrini/dapr-state-store-app/blob/main/dapr.state.service/dapr-state-deploy.yaml)

**Comment by sunzhaochang** [comment]
The dapr annotations in deployment yaml should be set to `.spec.template.metadata.annotations`, not `.metadata.annotations`. So the yaml should be: 
```
apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    app: dapr-state
  name: dapr-state
  namespace: default
spec:
  replicas: 1
  selector:
    matchLabels:
      app: dapr-state
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: dapr-state
      annotations:
        dapr.io/enabled: "true"
        dapr.io/app-id: "dapr-state"
        dapr.io/app-port: "80"
        dapr.io/enable-api-logging: "true"
    spec:
      containers:
      - image: venkateshsrini3/dapr-state-service
        name: dapr-state-service
        ports:
        - containerPort: 80
        env:
        - name: "State.Store"
          value: "redis-cache-store"
```
BTW, the warning log `Unable to get SA xxx` here is not printed correctly, and it will be fixed by #4901

**Comment by VenkateshSrini** [comment]
thanks that solved the issue
